p2p/discover/udp.go

```diff
@@ -369,20 +373,41 @@ func (t *udp) loop() {
 						p.errc <- nil
 						plist.Remove(el)
 					}
+					// Reset the continuous timeout counter (time drift detection)
+					contTimeouts = 0
 				}
 			}
 			r.matched <- matched
 
 		case now := <-timeout.C:
 			nextTimeout = nil
+
 			// Notify and remove callbacks whose deadline is in the past.
 			for el := plist.Front(); el != nil; el = el.Next() {
 				p := el.Value.(*pending)
 				if now.After(p.deadline) || now.Equal(p.deadline) {
 					p.errc <- errTimeout
 					plist.Remove(el)
+					contTimeouts++
 				}
 			}
+			// If we've accumulated too many timeouts, do an NTP time sync check
+			if contTimeouts > ntpThreshold {
+				go func() {
+					drift, err := sntpDrift(3)
+					switch {
+					case err != nil:
+						glog.V(logger.Warn).Infof("No UDP connectivity, maybe blocked by firewall? (%v)", err)
```

Comment:
I think we shouldn't warn in this case. The purpose of the NTP check is warning in case of clock issues.